pkg/cmd/task/task.go

```diff
@@ -115,10 +115,11 @@ func mesosHTTPClient(ctx api.Context, agentID string) (*httpcli.Client, error) {
 
 		for _, a := range agents {
 			if a.AgentInfo.ID.Value == agentID {
+
 				// PID format: Host@IP:Port
-				s1 := strings.Split(a.GetPID(), "@")
-				s2 := strings.Split(s1[1], ":")
-				url := fmt.Sprintf("http://%s:%s", s2[0], s2[1])
+				// TODO: AgentInfo provides a hostname and a port, investigate if they can be used instead.
+				ipPort := strings.Split(a.GetPID(), "@")
+				url := fmt.Sprintf("http://%s", ipPort)
```

Comment:
`http` probably won't work in strict mode clusters, you'll want `https` for that, right?

Reply:
This code block is getting a client for pure mesos clusters. Users (usually core team members), can do eg. `dcos config set core.mesos_master_url http://localhost:9090`, and then the CLI will use this cluster for Mesos related tasks. The URL creation logic was taken from our Python codebase: https://github.com/dcos/dcos-core-cli/blob/90c3bf3e1b3198c6ed06242cebd2998e1ea67b06/python/lib/dcos/dcos/mesos.py#L662

For DC/OS cluster it happens a few lines below, where we create a Mesos HTTP client with our custom RoundTripper, which is handling HTTPS if the cluster was setup accordingly when doing `dcos cluster setup`.